File: nexora/backend/src/main/java/com/nexora/model/Email.java
package com.nexora.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "emails",
    indexes = {
        @Index(name = "idx_user_category", columnList = "user_id, category"),
        @Index(name = "idx_user_priority", columnList = "user_id, priority"),
        @Index(name = "idx_user_received", columnList = "user_id, received_at")
    }
)
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Email {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    @JsonIgnore
    private User user;

    @Column(name = "gmail_message_id", unique = true, nullable = false)
    private String gmailMessageId;

    @Column(name = "gmail_thread_id")
    private String gmailThreadId;

    @Column(name = "sender_name")
    private String senderName;

    @Column(name = "sender_email", nullable = false)
    private String senderEmail;

    @Column(columnDefinition = "TEXT")
    private String subject;

    @Column(name = "body_snippet", columnDefinition = "TEXT")
    private String bodySnippet;

    @JsonIgnore
    @Column(name = "body_full", columnDefinition = "LONGTEXT")
    private String bodyFull;

    @Column(name = "received_at")
    private LocalDateTime receivedAt;

    @Column(name = "is_read")
    @Builder.Default
    private Boolean isRead = false;

    @Column(name = "has_attachments")
    @Builder.Default
    private Boolean hasAttachments = false;

    @Enumerated(EnumType.STRING)
    @Builder.Default
    private EmailCategory category = EmailCategory.UNCATEGORIZED;

    @Enumerated(EnumType.STRING)
    @Builder.Default
    private Priority priority = Priority.MEDIUM;

    @Column(name = "ai_summary", columnDefinition = "TEXT")
    private String aiSummary;

    @Column(name = "ai_action_items", columnDefinition = "JSON")
    private String aiActionItems;

    @Column(name = "deadline_detected")
    private LocalDateTime deadlineDetected;

    @Column(name = "is_deadline_added_to_calendar")
    @Builder.Default
    private Boolean isDeadlineAddedToCalendar = false;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @OneToMany(mappedBy = "email", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<EmailAction> actions;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }

    public enum EmailCategory {
        ASSIGNMENT, ATTENDANCE, HACKATHON, PLACEMENT, INTERNSHIP,
        MEETING, ANNOUNCEMENT, RESEARCH, FINANCE, PERSONAL,
        PROMOTIONAL, SPAM, UNCATEGORIZED
    }

    public enum Priority {
        HIGH, MEDIUM, LOW
    }
}

File: nexora/frontend/src/types/Email.ts
export type EmailCategory =
  | 'ASSIGNMENT' | 'ATTENDANCE' | 'HACKATHON' | 'PLACEMENT' | 'INTERNSHIP'
  | 'MEETING' | 'ANNOUNCEMENT' | 'RESEARCH' | 'FINANCE' | 'PERSONAL'
  | 'PROMOTIONAL' | 'SPAM' | 'UNCATEGORIZED';

export type Priority = 'HIGH' | 'MEDIUM' | 'LOW';

export type EmailReaction = 'NONE' | 'DONE' | 'IMPORTANT' | 'LATER' | 'IGNORE' | 'SNOOZED';

export type ActionType = 'REGISTER' | 'REPLY' | 'SUBMIT' | 'UPLOAD' | 'REVIEW' | 'ATTEND' | 'OTHER';

export interface ActionItem {
  id: number;
  actionType: ActionType;
  actionDescription: string;
  deadline?: string;
  isCompleted: boolean;
}

export interface Email {
  id: number;
  gmailMessageId: string;
  gmailThreadId?: string;
  senderName?: string;
  senderEmail: string;
  subject?: string;
  bodySnippet?: string;
  bodyFull?: string;
  receivedAt?: string;
  isRead: boolean;
  hasAttachments: boolean;
  category: EmailCategory;
  priority: Priority;
  reaction?: EmailReaction;
  aiSummary?: string;
  aiActionItems?: string;
  deadlineDetected?: string;
  isDeadlineAddedToCalendar: boolean;
  actions?: ActionItem[];
  createdAt?: string;
}

export interface EmailPage {
  content: Email[];
  totalElements: number;
  totalPages: number;
  number: number;
  size: number;
}
